main.cpp

```diff
@@ -7,42 +7,371 @@
 using namespace std;
 
 class Figure {
-public:
-	Figure() {}
-	virtual bool check_turn() { return true; }
+	int row, col;
+	int row_turn, col_turn;
+	char color;
+public:	
+	Figure(int st_row, int st_col, char st_color) {
+		row = st_row;
+		col = st_col;
+		color = st_color;
+		row_turn = col_turn = 0;
+	}
+	virtual bool check_turn(int dst_row, int dst_col, char dst_color) {
```

Comment:
it seems that this function should never have this behavior (just returning true) and should always be overwritten. Thus, why don't we make it abstract (i.e. add = 0 instead of { ... })? Also, it should be marked as const, unless it ever needs to change some of the class variables, but it probably should not.

Reply:
I don't understand what you are talking about.

Reply:
You can read, e.g. this: https://www.geeksforgeeks.org/pure-virtual-functions-and-abstract-classes/ to learn more about abstract classes and this: https://www.geeksforgeeks.org/const-member-functions-c/ to learn more about "const" functions